lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/group/GroupProviderConfig.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.group;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+
+public class GroupProviderConfig
+{
+    private boolean isCachingEnabled;
+
+    public boolean isCachingEnabled()
+    {
+        return this.isCachingEnabled;
+    }
+
+    @Config("group-provider.cache.enabled")
```

Comment:
since it is a part of the plugin then I think we can drop the prefix `group-provider`

Reply:
I actually made the prefix configurable. It makes the module slightly more complex, but there's added flexibility.